sdk/batch/batch/test/batchClient.spec.ts

```diff
@@ -147,8 +148,16 @@ describe("Batch Service", () => {
     });
 
     it("should get a pool reference successfully", async () => {
-      const result = await client.pool.get("nodesdktestpool1");
-      const metadata = result.metadata![0];
+      let result, metadata;
+      while (true) {
+        result = await client.pool.get("nodesdktestpool1");
+        metadata = result.metadata![0];
+        if (result.allocationState === "resizing") {
+          await wait(10000);
```

Comment:
You should not wait. You should probably poll the expected state and then timeout if it doesn't reach it in the specified time.

Reply:
@bgklein I missed your comment somehow. @xingwu1 did you consider updating the swagger to provide LROs instead? I think that would make calling the APIs much cleaner.